src/app/components/POCs/ProcessorEdit/ProcessorEdit.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useCallback, VoidFunctionComponent } from "react";
+import { useMachine } from "@xstate/react";
+import { PageSection, PageSectionVariants } from "@patternfly/react-core";
+import ProcessorMachine from "@app/components/POCs/ProcessorEdit/ProcessorMachine";
+import ProcessorEditHeader from "@app/components/POCs/ProcessorEditHeader/ProcessorEditHeader";
+import ProcessorCodeEditor from "@app/components/POCs/ProcessorCodeEditor/ProcessorCodeEditor";
+import { ProcessorTemplateSelector } from "@app/components/POCs/ProcessorTemplateSelector/ProcessorTemplateSelector";
+import { ProcessorTemplate } from "@app/components/POCs/ProcessorEdit/ProcessorTemplates";
+import DeployBlockedModal from "@app/components/POCs/ProcessorEdit/components/DeployBlockedModal";
+import { useTranslation } from "@rhoas/app-services-ui-components";
+import "./ProcessorEdit.css";
+
+export interface ProcessorEditProps {
+  /** List of processor templates */
+  processorTemplates: ProcessorTemplate[];
+  /** List of sink values to use as suggestions while editing */
+  sinkValuesSuggestions: string[];
+  /** Callback to create a processor */
+  createProcessor: (
+    data: {
+      processorName: string;
+      code: string;
+    },
+    onSuccess: () => void,
+    onError: () => void
+  ) => void;
+  /** Callback to cancel the creation flow */
+  onCancel: () => void;
+  /** Existing name of the processor */
+  existingProcessorName?: string;
+  /** Existing code of the processor */
+  existingCode?: string;
+}
```

Comment:
I've named the component `ProcessorEdit` because we'll use it not only for the processor creation but also for editing existing processors. I've only prepared a couple of things for the editing part (like adding `existingProcessorName` and `existingCode` props) but I've only focused on the creation process for now.

Also the `sinkValuesSuggestions` is only accepting a list of values for now. It will likely change when we'll fetch existing sinks from APIs.